### PR TITLE
New rotate_test - fix for #6

### PR DESCRIPTION
**Pull request by karel-m**

### CODE REVIEW

**Comment by karel-m** [comment]
```
NO_ASM (-DLTC_NO_ASM -O2, makefile IGNORE_SPEED=1)...
building...build  NO_ASM failed! printing gcc_2.txt now for convenience
ar: creating libtomcrypt.a
rotate_test.c: In function ‘rotate_test’:
rotate_test.c:11:4: warning: left shift count >= width of type [enabled by default]
rotate_test.c:44:4: warning: left shift count >= width of type [enabled by default]
rotate_test.c:142:4: warning: right shift count >= width of type [enabled by default]
rotate_test.c:207:4: warning: left shift count >= width of type [enabled by default]
rotate_test.c:240:4: warning: left shift count >= width of type [enabled by default]
rotate_test.c:338:4: warning: right shift count >= width of type [enabled by default]
```

All of them are rotate by `0`.

One of TTC_NO_ASM macros:
```
#define RORc(x, y) ( ((((ulong32)(x)&0xFFFFFFFFUL)>>(ulong32)((y)&31)) | \
        ((ulong32)(x)<<(ulong32)(32-((y)&31)))) & 0xFFFFFFFFUL)
```

Warning is probably caused by `(32-((y)&31))` part.

**Comment by karel-m** [comment]
So now it compiles without a warning, but rotating by zero fails on `LTC_NO_ASM` builds.

File: testprof/rotate_test.c [hunk 1]
```diff
@@ -204,7 +204,7 @@ int rotate_test(void)
    if (ROL64(L64, 62) != CONST64(0x04488cd115599de2)) { fprintf(stderr, "ROL64 62 failed\n"); err++; }
    if (ROL64(L64, 63) != CONST64(0x089119a22ab33bc4)) { fprintf(stderr, "ROL64 63 failed\n"); err++; }
    /* RORc */
-   if (RORc(L32, 0 ) != 0x12345678UL) { fprintf(stderr, "RORc 0  failed\n"); err++; }
+   if (RORc(L32, ze) != 0x12345678UL) { fprintf(stderr, "RORc 0  failed\n"); err++; }
```
Comment (sjaeckel) [inline testprof/rotate_test.c]: That's against the idea of the `RO*c()` macros! The number of bits to rotate should be a constant, not a variable.

Comment (karel-m) [inline testprof/rotate_test.c]: After we fix the failure(s) we can return it back.

**Comment by karel-m** [comment]
I guess that for `LTC_NO_ASM` we need something like
```
- (ulong32)(x)<<(ulong32)(32-((y)&31))
+ (ulong32)(x)<<(ulong32)((32-((y)&31))&31)
```

**Comment by karel-m** [comment]
👍  Please merge this one

**Pull request merged**